collections/src/main/scala/strawman/collection/immutable/ImmutableArray.scala

```diff
@@ -285,5 +319,14 @@ object ImmutableArray extends StrictOptimizedSeqFactory[ImmutableArray] {
       case that: ofUnit => unsafeArray.length == that.unsafeArray.length
       case _ => super.equals(that)
     }
+    override def slice(from: Int, until: Int): ImmutableArray[Unit] = {
+      // cant use
+      // new ofUnit(util.Arrays.copyOfRange[Unit](array, from, until)) - Unit is special and doesnt compile
+      // cant use util.Arrays.copyOfRange[Unit](repr, from, until) - Unit is special and doesnt compile
+      val lo = scala.math.max(from, 0)
+      val res = new Array[Unit](until-lo)
```

Comment:
This is also incorrect (it copies the `copyOfRange` behavior instead of the normal `slice` behavior).  Note that I didn't mark all the places this needs to be fixed.

Reply:
Hi @Ichoran 
I fixed this and all copyOfRange invocation. But for this ofUnit case, I am not sure whether we need have this case. Because Unit will be auto-boxed to BoxedUnit at runtime, and per my testing, the runtime implementation of slice is rely on definition of ofRef.

Do you know under which situation, the ofUnit representation will be generated and used?